react/__tests__/Arrows.test.tsx

```diff
@@ -0,0 +1,152 @@
+import React, { Fragment } from 'react'
+import { render, fireEvent } from '@vtex/test-tools/react'
+
+import { useSliderState } from '../components/SliderContext'
+import Arrow from '../components/Arrow'
+
+const mockGoForward = jest.fn()
+const mockGoBack = jest.fn()
+
+jest.mock('../components/SliderContext', () => ({
+  useSliderState: jest.fn(),
+}))
+
+jest.mock('../hooks/useSliderControls', () => ({
+  useSliderControls: () => ({
+    goForward: mockGoForward,
+    goBack: mockGoBack,
+  }),
+}))
+
+beforeEach(() => {
+  mockGoForward.mockClear()
+  mockGoBack.mockClear()
+  // eslint-disable-next-line prettier/prettier
+})
+
+describe('Accessibility', () => {
+  ;(useSliderState as jest.Mock).mockImplementation(() => ({
+    currentSlide: 0,
+    slidesPerPage: 5,
+    navigationStep: 5,
+  }))
+
+  it('should render with correct aria attributes in each arrow', () => {
+    const expectedLabelLeft = 'Previous Slide'
+    const expectedLabelRight = 'Next Slide'
+    const expectedContainerRole = 'button'
+
+    const { queryAllByRole, queryByLabelText } = render(
+      <Fragment>
+        <Arrow
+          infinite
+          arrowSize={25}
+          totalItems={20}
+          orientation="left"
+          controls="slider-items"
+        />
+        <Arrow
+          infinite
+          arrowSize={25}
+          totalItems={20}
+          orientation="right"
+          controls="slider-items"
+        />
+      </Fragment>
+    )
+
+    expect(queryAllByRole(expectedContainerRole)).toHaveLength(2)
+    expect(queryByLabelText(expectedLabelLeft)).toBeTruthy()
+    expect(queryByLabelText(expectedLabelRight)).toBeTruthy()
+  })
+})
+
+describe('Behavior upon interaction', () => {
+  ;(useSliderState as jest.Mock).mockImplementation(() => ({
+    currentSlide: 0,
+    slidesPerPage: 5,
+    navigationStep: 5,
+  }))
+
+  it('should go to the next page if right arrow is clicked', () => {
+    const { queryByTestId } = render(
+      <Arrow
+        infinite
+        arrowSize={25}
+        totalItems={20}
+        orientation="right"
+        controls="slider-items"
+      />
+    )
+
+    const rightArrow = queryByTestId('icon-caret-right')
+
+    if (!rightArrow) {
+      throw new Error('Could not find right arrow')
+    }
```

Comment:
You can use `getByTestId` that already throws an error if it can't find the element

Reply:
For sure!! Nice!